flink-runtime/src/main/java/org/apache/flink/runtime/util/ZooKeeperUtils.java

```diff
@@ -246,6 +249,38 @@ public static CuratorFrameworkWithUnhandledErrorListener startCuratorFramework(
                         .ensembleTracker(ensembleTracking)
                         .aclProvider(aclProvider);
 
+        if (configuration.contains(HighAvailabilityOptions.ZOOKEEPER_CLIENT_AUTHORIZATION)) {
+            Map<String, String> authMap =
+                    configuration.get(HighAvailabilityOptions.ZOOKEEPER_CLIENT_AUTHORIZATION);
+            List<AuthInfo> authInfos =
+                    authMap.entrySet().stream()
+                            .map(
+                                    entry ->
+                                            new AuthInfo(
+                                                    entry.getKey(),
+                                                    entry.getValue()
+                                                            .getBytes(
+                                                                    ConfigConstants
+                                                                            .DEFAULT_CHARSET)))
+                            .collect(Collectors.toList());
+            curatorFrameworkBuilder.authorization(authInfos);
+        }
+
+        if (configuration.contains(HighAvailabilityOptions.ZOOKEEPER_MAX_CLOSE_WAIT_MS)) {
+            curatorFrameworkBuilder.maxCloseWaitMs(
+                    (int)
+                            configuration
+                                    .get(HighAvailabilityOptions.ZOOKEEPER_MAX_CLOSE_WAIT_MS)
+                                    .toMillis());
```

Comment:
```suggestion
            final Duration maxCloseWaitDuration =
                    configuration.get(HighAvailabilityOptions.ZOOKEEPER_MAX_CLOSE_WAIT_MS);
            final long maxCloseWaitInMillis = maxCloseWaitDuration.toMillis();
            Preconditions.checkArgument(
                    maxCloseWaitInMillis <= Integer.MAX_VALUE,
                    "The passed value of {} for {} exceeded to valid range of {}ms.",
                    maxCloseWaitDuration,
                    HighAvailabilityOptions.ZOOKEEPER_MAX_CLOSE_WAIT_MS.key(),
                    Duration.ofMillis(Integer.MAX_VALUE));
            curatorFrameworkBuilder.maxCloseWaitMs((int) maxCloseWaitInMillis);
```
Can we add another safety net for the cast to fail early if (by any chance) an unexpected value is used?

Reply:
You have a point there. It appears to be a quite nitty. But on the other hand, it's also quite unlikely that a user would set such a timeout. Therefore, an exception wouldn't hurt. :shrug: